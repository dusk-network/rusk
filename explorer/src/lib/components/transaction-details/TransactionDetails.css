.transaction-details__list {
  display: grid;
  grid-template-columns: minmax(0, 1fr);
  font-size: 0.875rem;
  column-gap: 1.875rem;
  padding: 0 1.125rem;
}

.transaction-details__status,
.transaction-details__type {
  width: 100%;
}

.transaction-details__gas-used {
  width: 12.5rem;
}

.transaction-details__payload {
  padding: 0.625rem 0.875rem;
  line-height: 1.3125rem;
  width: 100%;
  overflow-x: auto;
}

.transaction-details__payload-switch {
  position: absolute;
  right: -3.8125rem;
}

.transaction-details__list .details-list__term:nth-last-child(-n + 4) {
  align-items: start;
  margin-bottom: 0.625rem;
  /*
    3.8125rem is the sum of the switch width plus the gap
    between the switch and the dt element
  */
  width: calc(100% - 3.8125rem);
}

.transaction-details__memo {
  padding: 0.625rem 0.875rem;
  line-height: 1.3125rem;
  width: 100%;
  overflow-x: auto;
}

.transaction-details__switch-wrapper {
  display: flex;
  align-items: center;
  flex: 1 100%;
  justify-content: space-between;
}

.transaction-details__list-anchor {
  font-family: var(--mono-font-family);
}

.transaction-details__blob-hashes {
  list-style-type: none;
  display: flex;
  flex-direction: column;
  gap: var(--default-gap);
}

/**
 * We need to add a margin conditionally as these components are a bit
 * odd as the list term has a padding, while the definition doesn't.
 * If we have more than one li element we need a margin equal to that
 * padding to have the term and the definition aligned on top.
 */
.transaction-details__blob-hashes:has(> li:nth-child(2)) {
  margin-top: 0.9375rem;
}

.transaction-details__blob-hash {
  display: inline-flex;
  align-items: center;
  gap: var(--small-gap);
}

.transaction-details__blob-hash pre {
  white-space: pre-line;
}

@media (min-width: 640px) {
  .transaction-details__header-heading {
    text-transform: none;
  }

  .transaction-details__list {
    display: inline-grid;
    grid-template-columns: minmax(0, auto) minmax(0, 3fr);
  }

  .transaction-details__payload {
    margin-top: 1rem;
  }

  .transaction-details__memo {
    margin-top: 0rem;
  }

  .transaction-details__payload-switch {
    position: relative;
    right: 0rem;
  }

  .transaction-details__status,
  .transaction-details__type {
    width: auto;
  }

  .transaction-details__list .details-list__term:nth-last-child(-n + 4) {
    align-items: start;
    margin-bottom: 0rem;
    width: auto;
  }
}
