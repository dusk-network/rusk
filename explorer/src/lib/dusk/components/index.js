export { default as Anchor } from "./anchor/Anchor.svelte";
export { default as AnchorButton } from "./anchor-button/AnchorButton.svelte";
export { default as Banner } from "./banner/Banner.svelte";
export { default as Badge } from "./badge/Badge.svelte";
export { default as Button } from "./button/Button.svelte";
export { default as Card } from "./card/Card.svelte";
export { default as CopyButton } from "./copy-button/CopyButton.svelte";
export { default as Icon } from "./icon/Icon.svelte";
export { default as NavList } from "./nav-list/NavList.svelte";
export { default as ProgressBar } from "./progress-bar/ProgressBar.svelte";
export { default as Rerender } from "./rerender/Rerender.svelte";
export { default as RelativeTime } from "./relative-time/RelativeTime.svelte";
export { default as Select } from "./select/Select.svelte";
export { default as Switch } from "./switch/Switch.svelte";
export { default as Textbox } from "./textbox/Textbox.svelte";
export { default as Tooltip } from "./tooltip/Tooltip.svelte";
export { default as Toast } from "./toast/Toast.svelte";
