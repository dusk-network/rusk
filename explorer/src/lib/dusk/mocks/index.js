export { default as IntersectionObserver } from "./IntersectionObserver";
export { default as MediaQueryList } from "./MediaQueryList";
export { default as MediaQueryListEvent } from "./MediaQueryListEvent";
