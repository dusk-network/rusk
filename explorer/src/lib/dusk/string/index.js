export { default as calculateAdaptiveCharCount } from "./calculateAdaptiveCharCount";
export { default as decodeHexString } from "./decodeHexString";
export { default as getRelativeTimeString } from "./getRelativeTimeString";
export { default as hexToBytes } from "./hexToBytes";
export { default as isValidHex } from "./isValidHex";
export { default as makeClassName } from "./makeClassName";
export { default as middleEllipsis } from "./middleEllipsis";
