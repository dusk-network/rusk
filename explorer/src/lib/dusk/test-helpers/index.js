export { default as changeMediaQueryMatches } from "./changeMediaQueryMatches";
export { default as mockReadableStore } from "./mockReadableStore";
export { default as renderWithSimpleContent } from "./renderWithSimpleContent";
export { default as renderWithSlots } from "./renderWithSlots";
export { default as SlotContent } from "./SlotContent.svelte";
