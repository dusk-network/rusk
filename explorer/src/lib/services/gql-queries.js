export const transactionFragment = `
fragment TransactionInfo on SpentTransaction {
	blockHash,
	blockHeight,
	blockTimestamp,
  err,
	gasSpent,
	id,
  tx {
    ${import.meta.env.VITE_FEATURE_BLOB_HASHES === "true" ? "blobHashes," : ""}
    callData {
      contractId,
      data,
      fnName
    },
    gasLimit,
    gasPrice,
    id,
    isDeploy,
    memo,
    txType,
    json
  }
}
`;

const blockFragment = `
${transactionFragment}
fragment BlockInfo on Block {
  header {
    hash,
    gasLimit,
    height,
    prevBlockHash,
    seed,
    stateHash,
    timestamp,
    version
  },
  fees,
  gasSpent,
  reward,
  transactions {...TransactionInfo}
}
`;

/** @param {string} address */
export const getFullMoonlightAccountHistoryQuery = (address) => ({
  query: `
    query GetFullMoonlightHistory($address: String!) {
      fullMoonlightHistory(address: $address) {
        json
      }
    }
  `,
  variables: { address },
});

/** @param {number} height */
export const getBlockHashQueryInfo = (height) => ({
  query: `
    query($height: Float!) { block(height: $height) { header { hash } } }
  `,
  variables: { height },
});

/** @param {string} id */
export const getBlockQueryInfo = (id) => ({
  query: `
    ${blockFragment}
    query($id: String!) { block(hash: $id) {...BlockInfo} }
  `,
  variables: { id },
});

/** @param {string} id */
export const getBlockDetailsQueryInfo = (id) => ({
  query: "query($id: String!) { block(hash: $id) { header { json } } }",
  variables: { id },
});

/** @param {number} amount */
export const getBlocksQueryInfo = (amount) => ({
  query: `
    ${blockFragment}
    query($amount: Int!) { blocks(last: $amount) {...BlockInfo} }
  `,
  variables: { amount },
});

/** @param {number} amount */
export const getLatestChainQueryInfo = (amount) => ({
  query: `
    ${blockFragment}
    query($amount: Int!) {
      blocks(last: $amount) {...BlockInfo},
      transactions(last: $amount) {...TransactionInfo}
    }
  `,
  variables: { amount },
});

/** @param {string} id */
export const getMempoolTx = (id) => ({
  query: "query($id: String!) { mempoolTx(hash: $id) { isDeploy } }",
  variables: { id },
});

/** @param {number} amount */
export const getTransactionsQueryInfo = (amount) => ({
  query: `
    ${transactionFragment}
    query($amount: Int!) { transactions(last: $amount) {...TransactionInfo} }
  `,
  variables: { amount },
});

/** @param {string} id */
export const getTransactionQueryInfo = (id) => ({
  query: `
    ${transactionFragment}
    query($id: String!) {
      tx(hash: $id) {
        ...TransactionInfo
        tx {
          json
        }
      }
    }
  `,
  variables: { id },
});

/** @param {string} id */
export const searchByHashQueryInfo = (id) => ({
  query: `
    query($id: String!) {
      block(hash: $id) { header { hash } },
      tx(hash: $id) { id }
    }
  `,
  variables: { id },
});
