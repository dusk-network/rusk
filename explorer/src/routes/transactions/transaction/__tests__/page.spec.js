import { afterAll, afterEach, describe, expect, it, vi } from "vitest";
import { cleanup, render } from "@testing-library/svelte";

import { resolveAfter } from "$lib/dusk/promise";
import { duskAPI } from "$lib/services";

import TransactionDetails from "../+page.svelte";

const marketDataSettleTime = vi.hoisted(() => {
  vi.useFakeTimers();
  vi.setSystemTime(new Date(2024, 4, 20));

  return 100;
});

vi.mock("$lib/services", async (importOriginal) => {
  /** @type {import("$lib/services")} */
  const original = await importOriginal();
  const { transformTransaction } = await import("$lib/chain-info");
  const { apiMarketData, gqlTransaction } = await import("$lib/mock-data");
  const { current_price: currentPrice, market_cap: marketCap } =
    apiMarketData.market_data;

  return {
    ...original,
    duskAPI: {
      ...original.duskAPI,
      getMarketData: () =>
        resolveAfter(marketDataSettleTime, { currentPrice, marketCap }),
      getTransaction: vi
        .fn()
        .mockResolvedValue(transformTransaction(gqlTransaction.tx)),
    },
  };
});

describe("Transaction Details", () => {
  afterEach(() => {
    cleanup();
    vi.clearAllMocks();
  });

  afterAll(() => {
    vi.useRealTimers();
    vi.doUnmock("$lib/services");
  });

  it("should render the Transaction details page", async () => {
    const { container } = render(TransactionDetails);

    expect(container.firstChild).toMatchSnapshot();

    expect(duskAPI.getTransaction).toHaveBeenCalledTimes(1);

    await vi.advanceTimersByTimeAsync(marketDataSettleTime);

    // snapshot with received data from APIs
    expect(container.firstChild).toMatchSnapshot();
  });
});
