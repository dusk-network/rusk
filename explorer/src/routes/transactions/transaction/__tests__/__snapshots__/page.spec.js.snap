// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Transaction Details > should render the Transaction details page 1`] = `
<section
  class="transaction"
>
  <div
    class="dusk-card data-card transaction-details"
  >
    <header
      class="data-card__header"
      slot="header"
    >
      <h1
        class="data-card__header-title"
      >
        Transaction Details
      </h1>
       
    </header>
     
    <div
      aria-label="Loading"
      class="dusk-progress-bar data-card__progress-bar"
      role="progressbar"
    >
      <div
        class="dusk-progress-bar__filler dusk-progress-bar__filler--undetermined"
      />
    </div>
    
     
  </div>
  
</section>
`;

exports[`Transaction Details > should render the Transaction details page 2`] = `
<section
  class="transaction"
>
  <div
    class="dusk-card data-card transaction-details"
  >
    <header
      class="data-card__header"
      slot="header"
    >
      <h1
        class="data-card__header-title"
      >
        Transaction Details
      </h1>
       
    </header>
     
    <div
      class="data-card__content"
    >
      <dl
        class="transaction-details__list"
      >
        <dt
          class="details-list__term"
        >
          <div
            class="details-list__term-layout"
          >
            <svg
              class="dusk-icon dusk-icon--size--default detail-list__help"
              data-tooltip-id="main-tooltip"
              data-tooltip-place="top"
              data-tooltip-text="The ID of the transaction"
              data-tooltip-type="info"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
            
             
            ID
          </div>
        </dt>
         
        <dd
          class="details-list__definition"
        >
          <a
            class="dusk-anchor dusk-anchor--on-surface transaction-details__list-anchor"
            href="/some-base-path/transactions/transaction?id=4877687c2dbf154248d3ddee9ba0d81e3431f39056f82a46819da041d4ac0e04"
          >
            4877687c2dbf154248d3ddee9...056f82a46819da041d4ac0e04
          </a>
           
          <div
            class="dusk-copy-button"
          >
            <button
              aria-label="Copy Address"
              class="dusk-button dusk-button--type--button dusk-button--variant--primary dusk-button--size--default dusk-icon-button dusk-copy-button"
              data-tooltip-id="main-tooltip"
              data-tooltip-text="Copy to clipboard"
              type="button"
            >
              <svg
                class="dusk-icon dusk-icon--size--default dusk-button__icon"
                role="graphics-symbol"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"
                />
              </svg>
              
               
              
            </button>
          </div>
           
        </dd>
        <dt
          class="details-list__term"
        >
          <div
            class="details-list__term-layout"
          >
            <svg
              class="dusk-icon dusk-icon--size--default detail-list__help"
              data-tooltip-id="main-tooltip"
              data-tooltip-place="top"
              data-tooltip-text="The block height this transaction belongs to"
              data-tooltip-type="info"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
            
             
            block height
          </div>
        </dt>
         
        <dd
          class="details-list__definition"
        >
          <a
            class="dusk-anchor dusk-anchor--on-surface transaction-details__list-anchor"
            href="/some-base-path/blocks/block?id=3c6e4018cfa86723e50644e33d3990bc27fc794f6b49fbf6290e4d308e07bd2d"
          >
            487,166
          </a>
        </dd>
        <dt
          class="details-list__term"
        >
          <div
            class="details-list__term-layout"
          >
            <svg
              class="dusk-icon dusk-icon--size--default detail-list__help"
              data-tooltip-id="main-tooltip"
              data-tooltip-place="top"
              data-tooltip-text="The date and time the transaction was created"
              data-tooltip-type="info"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
            
             
            timestamp
          </div>
        </dt>
         
        <dd
          class="details-list__definition"
        >
          <time
            class="dusk-relative-time transaction-details__list-timestamp"
            datetime="2024-04-16T06:39:09.000Z"
            slot="definition"
          >
            Tue, 16 Apr 2024 06:39:09 GMT (last month)
            
          </time>
        </dd>
        <dt
          class="details-list__term"
        >
          <div
            class="details-list__term-layout"
          >
            <svg
              class="dusk-icon dusk-icon--size--default detail-list__help"
              data-tooltip-id="main-tooltip"
              data-tooltip-place="top"
              data-tooltip-text="The sender of the transaction"
              data-tooltip-type="info"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
            
             
            from
          </div>
        </dt>
         
        <dd
          class="details-list__definition"
        >
          <a
            class="dusk-anchor dusk-anchor--on-surface transaction-details__list-anchor"
            href="/some-base-path/accounts/?key=kT5BamV2Xq6JpnoCq9ux7EmYRShdg4W1sLzyGFUqicmZTqxeMWLiTGpBWNhpjiPtKqvWFFmdxVBAjts3oEnZgPfuiyTRwDdASJQjTSy7wNhHdCeKtzExREcXZsmmMRLwKbXj"
          >
            kT5BamV2Xq6JpnoCq9ux7EmYR...HdCeKtzExREcXZsmmMRLwKbXj
          </a>
           
          <div
            class="dusk-copy-button"
          >
            <button
              aria-label="Copy Address"
              class="dusk-button dusk-button--type--button dusk-button--variant--primary dusk-button--size--default dusk-icon-button dusk-copy-button"
              data-tooltip-id="main-tooltip"
              data-tooltip-text="Copy to clipboard"
              type="button"
            >
              <svg
                class="dusk-icon dusk-icon--size--default dusk-button__icon"
                role="graphics-symbol"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"
                />
              </svg>
              
               
              
            </button>
          </div>
           
        </dd>
        
        <dt
          class="details-list__term"
        >
          <div
            class="details-list__term-layout"
          >
            <svg
              class="dusk-icon dusk-icon--size--default detail-list__help"
              data-tooltip-id="main-tooltip"
              data-tooltip-place="top"
              data-tooltip-text="The receiver of the transaction"
              data-tooltip-type="info"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
            
             
            to
          </div>
        </dt>
         
        <dd
          class="details-list__definition"
        >
          <a
            class="dusk-anchor dusk-anchor--on-surface transaction-details__list-anchor"
            href="/some-base-path/accounts/?key=mHWZo9qYUhqp2SEmtWN7EuDKFwVrjgdEyzpHdfLB6R9scRAq5EkUQyLB9fBfCGt1wjxfEpFxHq9MBPVVPY3Lk3JKnQLWZzj7UYAR4mGmeGQZwCaeCS8uA63ZPKnGpjiUnj"
          >
            mHWZo9qYUhqp2SEmtWN7EuDKF...GQZwCaeCS8uA63ZPKnGpjiUnj
          </a>
           
          <div
            class="dusk-copy-button"
          >
            <button
              aria-label="Copy Address"
              class="dusk-button dusk-button--type--button dusk-button--variant--primary dusk-button--size--default dusk-icon-button dusk-copy-button"
              data-tooltip-id="main-tooltip"
              data-tooltip-text="Copy to clipboard"
              type="button"
            >
              <svg
                class="dusk-icon dusk-icon--size--default dusk-button__icon"
                role="graphics-symbol"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"
                />
              </svg>
              
               
              
            </button>
          </div>
           
        </dd>
        
        <dt
          class="details-list__term"
        >
          <div
            class="details-list__term-layout"
          >
            <svg
              class="dusk-icon dusk-icon--size--default detail-list__help"
              data-tooltip-id="main-tooltip"
              data-tooltip-place="top"
              data-tooltip-text="The transaction amount"
              data-tooltip-type="info"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
            
             
            amount
          </div>
        </dt>
         
        <dd
          class="details-list__definition"
        >
          9,812.378912731 DUSK ($3,614.3897725045)
          
           
          
        </dd>
        
        <dt
          class="details-list__term"
        >
          <div
            class="details-list__term-layout"
          >
            <svg
              class="dusk-icon dusk-icon--size--default detail-list__help"
              data-tooltip-id="main-tooltip"
              data-tooltip-place="top"
              data-tooltip-text="The transaction type"
              data-tooltip-type="info"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
            
             
            type
          </div>
        </dt>
         
        <dd
          class="details-list__definition"
        >
          <div
            class="transaction-type"
          >
            <svg
              class="dusk-icon dusk-icon--size--large transaction-type__icon"
              data-tooltip-disabled="false"
              data-tooltip-id="main-tooltip"
              data-tooltip-place="top"
              data-tooltip-text="Public"
              data-tooltip-type="info"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M21 11C21 16.5 17.2 21.7 12 23C6.8 21.7 3 16.5 3 11V5L12 1L21 5V11M12 21C15.8 20 19 15.5 19 11.2V6.3L12 3.2L5 6.3V11.2C5 15.5 8.2 20 12 21M14.8 11H10.5V8.5C10.5 7.7 11.2 7.2 12 7.2S13.5 7.7 13.5 8.5V9H14.8V8.5C14.8 7.1 13.4 6 12 6S9.2 7.1 9.2 8.5V11C8.6 11 8 11.6 8 12.2V15.7C8 16.4 8.6 17 9.2 17H14.7C15.4 17 16 16.4 16 15.8V12.3C16 11.6 15.4 11 14.8 11Z"
              />
            </svg>
            
             
            <span
              class="dusk-badge dusk-badge--variant-neutral transaction-type__method-badge"
            >
              transfer
            </span>
             
          </div>
           
        </dd>
        <dt
          class="details-list__term"
        >
          <div
            class="details-list__term-layout"
          >
            <svg
              class="dusk-icon dusk-icon--size--default detail-list__help"
              data-tooltip-id="main-tooltip"
              data-tooltip-place="top"
              data-tooltip-text="The transaction status"
              data-tooltip-type="info"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
            
             
            status
          </div>
        </dt>
         
        <dd
          class="details-list__definition"
        >
          <span
            class="dusk-badge dusk-badge--variant-success transaction-status transaction-details__status explorer-badge"
          >
            success
          </span>
        </dd>
        <dt
          class="details-list__term"
        >
          <div
            class="details-list__term-layout"
          >
            <svg
              class="dusk-icon dusk-icon--size--default detail-list__help"
              data-tooltip-id="main-tooltip"
              data-tooltip-place="top"
              data-tooltip-text="The fee paid for the transaction"
              data-tooltip-type="info"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
            
             
            transaction fee
          </div>
        </dt>
         
        <dd
          class="details-list__definition"
        >
          0.000290766 DUSK ($0.0001071037)
          
           
          
           
        </dd>
        <dt
          class="details-list__term"
        >
          <div
            class="details-list__term-layout"
          >
            <svg
              class="dusk-icon dusk-icon--size--default detail-list__help"
              data-tooltip-id="main-tooltip"
              data-tooltip-place="top"
              data-tooltip-text="The transaction gas price"
              data-tooltip-type="info"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
            
             
            gas price
          </div>
        </dt>
         
        <dd
          class="details-list__definition"
        >
          0.000000001 DUSK ($0.0000000004)
          
           
          
           
        </dd>
        <dt
          class="details-list__term"
        >
          <div
            class="details-list__term-layout"
          >
            <svg
              class="dusk-icon dusk-icon--size--default detail-list__help"
              data-tooltip-id="main-tooltip"
              data-tooltip-place="top"
              data-tooltip-text="The transaction gas limit"
              data-tooltip-type="info"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
            
             
            transaction gas limit
          </div>
        </dt>
         
        <dd
          class="details-list__definition"
        >
          500000000
        </dd>
        <dt
          class="details-list__term"
        >
          <div
            class="details-list__term-layout"
          >
            <svg
              class="dusk-icon dusk-icon--size--default detail-list__help"
              data-tooltip-id="main-tooltip"
              data-tooltip-place="top"
              data-tooltip-text="The amount of gas spent generating the transaction"
              data-tooltip-type="info"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
            
             
            gas spent
          </div>
        </dt>
         
        <dd
          class="details-list__definition"
        >
          290766
        </dd>
        <dt
          class="details-list__term"
        >
          <div
            class="details-list__term-layout"
          >
            <svg
              class="dusk-icon dusk-icon--size--default detail-list__help"
              data-tooltip-id="main-tooltip"
              data-tooltip-place="top"
              data-tooltip-text="The number of transactions sent from this address"
              data-tooltip-type="info"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
            
             
            nonce
          </div>
        </dt>
         
        <dd
          class="details-list__definition"
        >
          156
        </dd>
        
        <dt
          class="details-list__term"
        >
          <div
            class="details-list__term-layout"
          >
            <svg
              class="dusk-icon dusk-icon--size--default detail-list__help"
              data-tooltip-id="main-tooltip"
              data-tooltip-place="top"
              data-tooltip-text="Transaction reference and additional notes"
              data-tooltip-type="info"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
            
             
            <div
              class="transaction-details__switch-wrapper"
            >
              memo
          
              <div
                aria-checked="false"
                aria-disabled="false"
                class="dusk-switch transaction-details__payload-switch dusk-switch--on-surface"
                role="switch"
                tabindex="0"
              />
            </div>
             
          </div>
        </dt>
         
        <dd
          class="details-list__definition"
        >
          some notes about the transaction
          
           
        </dd>
        <dt
          class="details-list__term"
        >
          <div
            class="details-list__term-layout"
          >
            <svg
              class="dusk-icon dusk-icon--size--default detail-list__help"
              data-tooltip-id="main-tooltip"
              data-tooltip-place="top"
              data-tooltip-text="The transaction payload"
              data-tooltip-type="info"
              role="graphics-symbol"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
            
             
            <div
              class="transaction-details__switch-wrapper"
            >
              payload
          
              <div
                aria-checked="false"
                aria-disabled="false"
                class="dusk-switch transaction-details__payload-switch dusk-switch--on-surface"
                role="switch"
                tabindex="0"
              />
            </div>
          </div>
        </dt>
         
        <dd
          class="details-list__definition"
        >
          
        </dd>
      </dl>
    </div>
    
     
  </div>
  
</section>
`;
